### PR TITLE
[MM-44486] Use proper ICE server configs

### PR DESCRIPTION
**Pull request by streamer45**
#### Summary

Changing `ServerConfig.ICEServers` to be a list of proper ICE server config, potentially supporting TURN servers as well, which need credentials.

**Note**

This is a breaking change but it's one we can afford to make on this side. I've also added a backwards compatible unmarshaler for the config so that we can still support the previous format, just in case.

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-44486

### CODE REVIEW

File: service/rtc/config.go [hunk 1]
```diff
@@ -55,3 +61,99 @@ func (c SessionConfig) IsValid() error {
...
+		}
+	}
+	return nil
+}
+
+func (s ICEServers) getSTUN() string {
+	for _, cfg := range s {
+		for _, u := range cfg.URLs {
+			if strings.HasPrefix(u, "stun:") {
+				return u
```
Comment (cpoile) [inline service/rtc/config.go:102]: I guess that's expected, to return only the first?

**Review by cpoile — APPROVED** [review]
Cool. I'm assuming there's no breaking changing to client config, so we don't have to worry about that. :)

File: service/rtc/config.go (hunk 1 above)
Comment (streamer45) [reply service/rtc/config.go:102]: Yes, we just need one. We could randomize but not a big deal at the moment.

> Cool. I'm assuming there's no breaking changing to client config, so we don't have to worry about that. :)

On the client, I am still finishing up the PRs but we'll be adding a new ICEServerConfigs field along with the existing ICEServers which we'll deprecate. Will have PRs up tomorrow.

**Comment by streamer45** [comment]
> Cool. I'm assuming there's no breaking changing to client config, so we don't have to worry about that. :)

On the client, I am still finishing up the PRs but we'll be adding a new ICEServerConfigs field along with the existing ICEServers which we'll deprecate. Will have PRs up tomorrow.

**Pull request merged**